crates/holochain/tests/tests/conductor/mod.rs

```diff
@@ -0,0 +1,341 @@
+        Some(manifest_properties.clone())
+    );
+    assert_eq!(
+        installed_app_role_2.dna.modifiers.origin_time,
+        Some(manifest_origin_time)
+    );
+    assert_eq!(
+        installed_app_role_2.dna.modifiers.quantum_time,
+        Some(manifest_quantum_time)
+    );
+
+    //- Test that modifier fields that are None in the modifiers map do not overwrite existing
+    //  modifiers from the manifest
+    let custom_modifiers = DnaModifiersOpt::default();
+
+    let role_settings = (
+        role_name_1.clone(),
+        RoleSettings::Provisioned {
+            membrane_proof: Default::default(),
+            modifiers: Some(custom_modifiers.clone()),
+        },
+    );
+
+    conductor
+        .clone()
+        .install_app_bundle(InstallAppPayload {
+            agent_key: None,
+            source: AppBundleSource::Bundle(bundle.clone()),
+            installed_app_id: Some("app_2".into()),
+            network_seed: None,
+            roles_settings: Some(HashMap::from([role_settings])),
+            ignore_genesis_failure: false,
+            allow_throwaway_random_agent_key: true,
+        })
+        .await
+        .unwrap();
+
+    let manifest = conductor
+        .get_app_info(&"app_2".to_string())
+        .await
+        .unwrap()
+        .unwrap()
+        .manifest;
+
+    let installed_app_role_1 = manifest
+        .app_roles()
+        .into_iter()
+        .find(|r| r.name == role_name_1)
+        .unwrap();
+
+    // Check that the modifers have been set correctly
+    assert_eq!(
+        installed_app_role_1.dna.modifiers.network_seed,
+        Some(manifest_network_seed.clone())
+    );
+    assert_eq!(
+        installed_app_role_1.dna.modifiers.properties,
+        Some(manifest_properties.clone())
+    );
+    assert_eq!(
+        installed_app_role_1.dna.modifiers.origin_time,
+        Some(manifest_origin_time)
+    );
+    assert_eq!(
+        installed_app_role_1.dna.modifiers.quantum_time,
+        Some(manifest_quantum_time)
+    );
+
+    //- Check that installing with modifiers for a non-existent role fails
+    let role_settings = (
+        "unknown role name".into(),
+        RoleSettings::Provisioned {
+            membrane_proof: Default::default(),
+            modifiers: Some(custom_modifiers.clone()),
+        },
+    );
+
+    let result = conductor
+        .clone()
+        .install_app_bundle(InstallAppPayload {
+            agent_key: None,
+            source: AppBundleSource::Bundle(bundle.clone()),
+            installed_app_id: Some("app_3".into()),
+            network_seed: Some("final seed".into()),
+            roles_settings: Some(HashMap::from([role_settings])),
+            ignore_genesis_failure: false,
+            allow_throwaway_random_agent_key: true,
+        })
+        .await;
+
+    assert!(result.is_err());
+
+    //- Check that if providing a membrane proof in the role settings for an app with `allow_deferred_memproofs`
```

Comment:
This is a different topic to the modifiers, even more important that this goes in its own test